src/device_memory_tracker.h

```diff
@@ -0,0 +1,265 @@
+
+#pragma once
+
+#include <cupti.h>
```

Comment:
Does this work on a CPU only build? Or a container with no GPU driver? Might need to be protected `#ifdef TRITON_ENABLE_GPU`